pinot-common/src/main/java/org/apache/pinot/common/messages/SegmentReloadMessage.java

```diff
@@ -49,6 +50,13 @@ public SegmentReloadMessage(@Nonnull String tableNameWithType, @Nullable String
     znRecord.setBooleanField(FORCE_DOWNLOAD_KEY, forceDownload);
   }
 
+  public SegmentReloadMessage(@Nonnull String tableNameWithType, @Nullable String segmentName, boolean forceDownload,
+      int parallelism) {
```

Comment:
I know this is draft PR

Just curious on the approach. How does the user specify parallelism parameter ? Can we use semaphore like we do in refresh (which is also delivered to server via helix message) ?

Reply:
@suddendust if this is a draft PR, can you please add WIP in the title? thanks

Reply:
@siddharthteotia After some discussion, the conclusion is that there is no point in having yet another param to control the parallelism. Instead, using the existing `_refreshThreadsSemaphore` for this makes sense as reload and refresh are so similar. For each available permit, we reload a segment asynchronously.

Reply:
Yes, sound good